File: src/Irrigarten/Dice.java
package Irrigarten;
import java.util.Random;

 public class Dice {
    static private final int MAX_USES = 5;
    static private final float MAX_INTELLIGENCE = 10.0f;
    static private final float MAX_STRENGTH = 10.0f;
    static private final float RESURRECT_PROB = 0.3f;
    static private final int WEAPONS_REWARD = 2;
    static private final int SHIELDS_REWARD = 3;
    static private final int HEALTH_REWARD = 5;
    static private final int MAX_ATTACK = 3;
    static private final int MAX_SHIELD = 2;
    static private Random generator = new Random();
    
     public static int randomPos(int max){
        return generator.nextInt(max);
    }

    public static int whoStarts(int nplayers){
        return generator.nextInt(nplayers);
    }
     public static float randomIntelligence(){
        return generator.nextInt((int)MAX_INTELLIGENCE) + generator.nextFloat();

    }
     public static float randomStreangth(){
        return generator.nextInt((int)MAX_STRENGTH) + generator.nextFloat();
    }

     public static boolean resurrectPlayer(){
        if (generator.nextDouble() > RESURRECT_PROB){
            return false;
        }
        else{
            return true;
        }
    }

     public static int weaponsReward(){
        return generator.nextInt(WEAPONS_REWARD+1);
    }
     public static int shieldsReward(){
        return generator.nextInt(SHIELDS_REWARD);
    }
     public static int healthReward(){
        return generator.nextInt(HEALTH_REWARD);
    }

     public static float weaponPower(){
        return generator.nextInt(MAX_ATTACK) + generator.nextFloat();
    }

     public static float shieldPower(){
        return generator.nextInt(MAX_SHIELD) + generator.nextFloat();
    }
     public static int usesLeft(){
        return generator.nextInt(MAX_USES);
    }
     public static float intensity(float competence){
        return competence * generator.nextFloat();
    }
     public static boolean discardElement(int usesLeft){
        float prob = usesLeft/MAX_USES;

        return generator.nextFloat() <= prob;

    }
}

File: JUEGO-RUBY/irrgarten/Dice.rb
module Irrgarten
    class Dice
        @@MAX_USES = 5
        @@MAX_INTELLIGENCE = 10.0
        @@MAX_STRENGTH = 10.0
        @@RESURRECT_PROB = 0.3
        @@WEAPONS_REWARD = 2
        @@SHIELDS_REWARD = 3
        @@HEALTH_REWARD = 5
        @@MAX_ATTACK = 3.0
        @@MAX_SHIELD = 2.0
        @@generator = Random.new()
        
        def self.random_pos(max)
            return @@generator.rand(max)
        end

        def self.who_starts(nplayers)
            return @@generator.rand(nplayers)
        end

        def self.random_intelligence()
            return @@generator.rand(@@MAX_INTELLIGENCE)
        end
        
        def self.random_strength()
            return @@generator.rand(@@MAX_STRENGTH)
        end

        def self.resurrect_player()
            if @@generator.rand(0.0..1.0) > @@RESURRECT_PROB then
                return false
            else
                return true
            end
        end

        def self.weapons_reward()
            return @@generator.rand(0..@@WEAPONS_REWARD)
        end
        
        def self.shields_reward()
            return @@generator.rand(@@SHIELDS_REWARD)
        end

        def self.health_reward()
            return @@generator.rand(@@HEALTH_REWARD)
        end

        def self.weapon_power()
            return @@generator.rand(0..@@MAX_ATTACK)
        end

        def self.shield_power()
            return @@generator.rand(0..@@MAX_SHIELD)
        end

        def self.uses_left()
            return @@generator.rand(@@MAX_USES)
        end
        
        def self.intensity(competence)
            return @@generator.rand(0.0 ..1.0) * competence
        end

        def self.discard_element(uses_left)
            prob = uses_left.to_f/@@MAX_USES
            return (@@generator.rand(0.0..1.0)>= prob)
        end
        def self.next_step(preference,valid_moves,intelligence)
            prob = intelligence/@@MAX_INTELLIGENCE
            if @@generator.rand(0.0..1.0)<= prob
                return preference
            else
                return valid_moves[self.random_pos(valid_moves.size)]
            end
        end
    end
end


    



